### PR TITLE
RIA-4932 Decision - Appellant notification

### PR DESCRIPTION
**Pull request by burale**
### JIRA link (if applicable) ###
[RIA-4932](https://tools.hmcts.net/jira/browse/RIA-4932)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by gavinslilley — APPROVED** [review]

**Review by srinivasuluputcha — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/email/AppellantAppealOutcomePersonalisationEmail.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return
+            ImmutableMap
+                .<String, String>builder()
+                .putAll(customerServicesProvider.getCustomerServicesPersonalisation())
+                .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("homeOfficeReferenceNumber", asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("ariaListingReference", asylumCase.read(ARIA_LISTING_REFERENCE, String.class).orElse(""))
+                .put("appellantGivenNames", asylumCase.read(AsylumCaseDefinition.APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+                .put("appellantFamilyName", asylumCase.read(AsylumCaseDefinition.APPELLANT_FAMILY_NAME, String.class).orElse(""))
+                .put("14 or 28 days after judge uploads decisions and reasons document", "14 days")
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/email/AppellantAppealOutcomePersonalisationEmail.java:75]: @burale 14 days is still hard-coded here

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/sms/AppellantAppealOutcomePersonalisationSms.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Override
+    public Map<String, String> getPersonalisation(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        return
+            ImmutableMap
+                .<String, String>builder()
+                .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("14 or 28 days after judge uploads decisions and reasons document", "14 days")
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/sms/AppellantAppealOutcomePersonalisationSms.java:66]: @burale 14 days is still hard-coded here.

**Review by bradsorour — COMMENTED** [review]
@burale you also need a functional test to test for the SMS notifications going out.

`_APPEAL_OUTCOME_AIP_APPELLANT_SMS`

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/email/AppellantAppealOutcomePersonalisationEmail.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return caseId + "_APPEAL_OUTCOME_AIP_APPELLANT_EMAIL";
+    }
+
+    @Override
+    public Map<String, String> getPersonalisation(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        YesOrNo appellantInUK = asylumCase.read(AsylumCaseDefinition.APPELLANT_IN_UK, YesOrNo.class).orElse(YesOrNo.NO);
+
+        String period = appellantInUK == YesOrNo.YES ? "14 days" : "28 days";
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/email/AppellantAppealOutcomePersonalisationEmail.java:69]: @burale should this be `14` or `14 days`

File: src/functionalTest/resources/scenarios/RIA-4932-aip-appeal-outcome-allowed.json [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                "subscriber": "appellant",
+                "email": "{$TEST_CITIZEN_USERNAME}",
+                "wantsEmail": "Yes",
+                "mobileNumber": "{$TEST_CITIZEN_MOBILE}",
+                "wantsSms": "Yes"
+              }
+            }
+          ],
+          "ariaListingReference": "LP/12345/2019",
+          "isDecisionAllowed": "dismissed",
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-4932-aip-appeal-outcome-allowed.json:29]: @burale this `dismissed` decision doesn't tie in with the test description on line 2.

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/email/AppellantAppealOutcomePersonalisationEmail.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        return caseId + "_APPEAL_OUTCOME_AIP_APPELLANT_EMAIL";
+    }
+
+    @Override
+    public Map<String, String> getPersonalisation(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        YesOrNo appellantInUK = asylumCase.read(AsylumCaseDefinition.APPELLANT_IN_UK, YesOrNo.class).orElse(YesOrNo.NO);
+
+        String period = appellantInUK == YesOrNo.YES ? "14" : "28";
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/email/AppellantAppealOutcomePersonalisationEmail.java:69]: @burale this is incorrect when looking at your GovNotify template id `1d3266fe-ec61-4bd2-a7de-50092ec557ff`

If you have as `14` then your GovNotify template needs to change. Right now, the email going out is going to read:

`If you think the decision to dismiss your appeal is wrong, you have until 14 after judge uploads decisions and reasons document)) to ask for permission to appeal to the Upper Tribunal.`

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/sms/AppellantAppealOutcomePersonalisationSms.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        return caseId + "_APPEAL_OUTCOME_AIP_APPELLANT_SMS";
+    }
+
+    @Override
+    public Map<String, String> getPersonalisation(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        YesOrNo appellantInUK = asylumCase.read(AsylumCaseDefinition.APPELLANT_IN_UK, YesOrNo.class).orElse(YesOrNo.NO);
+
+        String period = appellantInUK == YesOrNo.YES ? "14" : "28";
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/sms/AppellantAppealOutcomePersonalisationSms.java:65]: @burale same as above applies to your sms template.

**Review by bradsorour — COMMENTED** [review]
Some issues still to be fixed.

**Review by bradsorour — APPROVED** [review]
LGTM

File: src/functionalTest/resources/scenarios/RIA-4932-aip-appeal-outcome-allowed.json [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        "reference": "1111_APPEAL_OUTCOME_AIP_APPELLANT_EMAIL",
+        "recipient": "{$TEST_CITIZEN_USERNAME}",
+        "subject": "Immigration and Asylum appeal: Your appeal has been decided",
+        "body": [
+          "PA/12345/2019",
+          "LP/12345/2019",
+          "A1234567",
+          "Talha Awan",
+          "{$iaAipFrontendUrl}"
+        ]
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-4932-aip-appeal-outcome-allowed.json:83]: @burale your functional tests need to check for the dynamic placeholders in the template. You're missing the check for `14 days` and also `customer service telephone and email`

File: src/functionalTest/resources/scenarios/RIA-4932-aip-appeal-outcome-dismissed.json [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        "reference": "2222_APPEAL_OUTCOME_AIP_APPELLANT_EMAIL",
+        "recipient": "{$TEST_CITIZEN_USERNAME}",
+        "subject": "Immigration and Asylum appeal: Your appeal has been decided",
+        "body": [
+          "PA/12345/2019",
+          "LP/12345/2019",
+          "A1234567",
+          "Talha Awan",
+          "{$iaAipFrontendUrl}"
+        ]
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-4932-aip-appeal-outcome-dismissed.json:83]: @burale same as above....

File: src/functionalTest/resources/scenarios/RIA-4932-aip-ooc-appeal-outcome-allowed.json [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        "reference": "3333_APPEAL_OUTCOME_AIP_APPELLANT_EMAIL",
+        "recipient": "{$TEST_CITIZEN_USERNAME}",
+        "subject": "Immigration and Asylum appeal: Your appeal has been decided",
+        "body": [
+          "PA/12345/2019",
+          "LP/12345/2019",
+          "A1234567",
+          "Talha Awan",
+          "{$iaAipFrontendUrl}"
+        ]
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-4932-aip-ooc-appeal-outcome-allowed.json:83]: @burale same as above...

File: src/functionalTest/resources/scenarios/RIA-4932-ooc-aip-appeal-outcome-dismissed.json [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+        "reference": "4444_APPEAL_OUTCOME_AIP_APPELLANT_EMAIL",
+        "recipient": "{$TEST_CITIZEN_USERNAME}",
+        "subject": "Immigration and Asylum appeal: Your appeal has been decided",
+        "body": [
+          "PA/12345/2019",
+          "LP/12345/2019",
+          "A1234567",
+          "Talha Awan",
+          "{$iaAipFrontendUrl}"
+        ]
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-4932-ooc-aip-appeal-outcome-dismissed.json:83]: @burale same as above...

**Review by bradsorour — APPROVED** [review]
@burale please ensure your functional tests check for the all dynamic placeholder content in the templates going forward.

**Pull request merged**